crates/sui-core/src/authority/shared_object_congestion_tracker.rs

```diff
@@ -149,20 +160,16 @@ impl SharedObjectCongestionTracker {
             return None;
         }
         let start_cost = self.compute_tx_start_at_cost(&shared_input_objects);
+        let end_cost = start_cost.saturating_add(tx_cost);
 
-        // Allow tx if it's within budget.
-        if start_cost.saturating_add(tx_cost) <= self.max_accumulated_txn_cost_per_object_in_commit
-        {
-            return None;
-        }
-
-        // Allow over-budget tx if it's not above the overage limit.
-        if start_cost <= self.max_accumulated_txn_cost_per_object_in_commit
-            && start_cost.saturating_add(tx_cost)
-                <= self
-                    .max_accumulated_txn_cost_per_object_in_commit
-                    .saturating_add(self.max_txn_cost_overage_per_object_in_commit)
-        {
+        // Allow tx if it's within configured limits.
+        let burst_limit = self
+            .max_accumulated_txn_cost_per_object_in_commit
+            .saturating_add(self.allowed_txn_cost_overage_burst_per_object_in_commit);
+        let absolute_limit = self
+            .max_accumulated_txn_cost_per_object_in_commit
+            .saturating_add(self.max_txn_cost_overage_per_object_in_commit);
+        if start_cost <= burst_limit && end_cost <= absolute_limit {
```

Comment:
In a scenario where there is a consistent congestion, after the absolute_limit is hit and it starts to defer transactions, it won't wait until all the debts are paid off and then admit new transactions, but as soon as `start_cost <= burst_limit` is true, then it will admit new transactions, is that correct?

Reply:
Correct, we decided that method would be the hardest to attack. 

PS. `absolute_limit` would never be hit with current settings, which have `self.max_txn_cost_overage_per_object_in_commit` set to `u64::MAX`